ui/src/components/credentials/IssuanceMethodForm.tsx

```diff
@@ -105,6 +98,12 @@ export function IssuanceMethodForm({
           ) {
             setIssuanceMethod({ ...allValues, linkExpirationTime: undefined });
           } else {
+            const parsedDid = didParser.safeParse(changedValues);
+
+            if (parsedDid.success) {
+              onChangeDid(parsedDid.data.did);
+            }
```

Comment:
Maybe we can reuse the `issuanceMethodFormDataParser` we already have instead of creating a new parser just for the DID?

```typescript
const parsedIssuanceMethodFormData =
  issuanceMethodFormDataParser.safeParse(allValues);
if (
  parsedIssuanceMethodFormData.success &&
  parsedIssuanceMethodFormData.data.type === "directIssue"
) {
  onChangeDid(parsedIssuanceMethodFormData.data.did);
}
```

Reply:
We have to do the parser to the `changedValues` to not change the URL with other changes

Reply:
Ok, let's at least make the parser a first-class citizen and compose it in the `IssuanceMethodFormData` as we do with `LinkExpiration` and `linkExpirationParser`:

```diff
diff --git a/ui/src/adapters/parsers/forms.ts b/ui/src/adapters/parsers/forms.ts
index 5f495d8..4365c4b 100644
--- a/ui/src/adapters/parsers/forms.ts
+++ b/ui/src/adapters/parsers/forms.ts
@@ -76,15 +76,24 @@ const linkExpirationParser = getStrictParser<LinkExpiration>()(
   })
 );
 
+interface Did {
+  did?: string;
+}
+
+export const didParser = getStrictParser<Did>()(
+  z.object({
+    did: z.string().optional(),
+  })
+);
+
 export type IssuanceMethodFormData =
   | (LinkExpiration & {
       linkMaximumIssuance?: number;
       type: "credentialLink";
     })
-  | {
-      did?: string;
+  | (Did & {
       type: "directIssue";
-    };
+    });
 
 const issuanceMethodFormDataParser = getStrictParser<IssuanceMethodFormData>()(
   z.union([
@@ -94,10 +103,11 @@ const issuanceMethodFormDataParser = getStrictParser<IssuanceMethodFormData>()(
         type: z.literal("credentialLink"),
       })
     ),
-    z.object({
-      did: z.string().optional(),
-      type: z.literal("directIssue"),
-    }),
+    didParser.and(
+      z.object({
+        type: z.literal("directIssue"),
+      })
+    ),
   ])
 );
 
@@ -196,14 +206,6 @@ export const linkExpirationDateParser = getStrictParser<{
   })
 );
 
-export const didParser = getStrictParser<{
-  did: string;
-}>()(
-  z.object({
-    did: z.string(),
-  })
-);
-
 // Serializers
 
 export function serializeCredentialLinkIssuance({
diff --git a/ui/src/components/credentials/IssuanceMethodForm.tsx b/ui/src/components/credentials/IssuanceMethodForm.tsx
index 70247a1..40eee03 100644
--- a/ui/src/components/credentials/IssuanceMethodForm.tsx
+++ b/ui/src/components/credentials/IssuanceMethodForm.tsx
@@ -37,7 +37,7 @@ export function IssuanceMethodForm({
   onSubmit,
 }: {
   initialValues: IssuanceMethodFormData;
-  onChangeDid: (did: string) => void;
+  onChangeDid: (did?: string) => void;
   onSubmit: (values: IssuanceMethodFormData) => void;
 }) {
   const env = useEnvContext();
```

Reply:
I've pushed a change, we can remove `linkExpirationDateParser` as well